### PR TITLE
[chore] Bump Go to 1.20

### LINKED ISSUE #474: Bump Go to 1.20
Both the repo to use Go 1.20 as OTel Go and Collector already depend on Go 1.20.

Also make sure to update https://github.com/open-telemetry/opentelemetry-go-build-tools/blob/41b2c0540f563c5eec30c6d59f07277d198e0748/crosslink/cmd/root.go#L146

### PR DESCRIPTION
**Pull request by webstradev**
Fixes #474 

I updated all the go mods and corresponding tests to 1.20

### CODE REVIEW

**Review by bryan-aguilar — APPROVED** [review]
Not sure why the GH jobs are hung, I also can't find windows-latest in the lint matrix for `ci.yml`. So I don't know where that is coming from.

**Comment by bryan-aguilar** [comment]
Weird, looks like the requirement is coming from this PR. https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/466

This seems like such an odd behavior from GitHub adding this requirement to a separate PR before the originating PR has been merged. Looks like we will have to wait for #466 to be merged.

**Comment by pellared** [comment]
@bryan-aguilar can you review #466? It awaits a review from someone who is not a Splunk employee.

**Review by pellared — APPROVED** [review]

**Pull request merged**
